tests/test_system.py

```diff
@@ -0,0 +1,210 @@
+import pytest
+
+from aubreylib import system
+
+
+class TestGetFileSystem():
+
+    location_tuple = ('file://disk2/',
+                      'http://unt.edu/disk2/')
+
+    @mock.patch('os.path.exists')
+    def test_file_local_to_server(self, mocked_exists):
+        """Locate a file when file_system starts with 'file://'"""
+        mocked_exists.return_value = True
+        path, location = system.get_file_system('metapthx',
+                                                'web/4.jpg',
+                                                self.location_tuple)
+        expected = ('web/4.jpg', '/disk2/')
+        assert (path, location) == expected
+
+    @mock.patch('os.path.exists')
+    def test_file_local_to_server_file_in_path(self, mocked_exists):
+        """Locate a system file when system and path both start
+        with 'file://'
+        """
+        mocked_exists.return_value = True
+        path, location = system.get_file_system('metapthx',
+                                                'file://web/4.jpg',
+                                                self.location_tuple)
+        expected = ('/disk2/me/ta/pt/hx/metapthx/web/4.jpg', '/disk2/')
+        assert (path, location) == expected
+
+    @mock.patch('os.path.exists')
+    @mock.patch('httplib.HTTP')
+    def test_file_at_http_url(self, MockedHTTP, mocked_exists):
+        """Locate a file on another server via http URL."""
+        MockedHTTP.return_value.getreply.return_value = (200, '', '')
+        mocked_exists.return_value = False
+        path, location = system.get_file_system('metapthx',
+                                                '/me/ta/pt/hx/metapthx/web/4.jpg',
+                                                self.location_tuple)
+        expected = ('http://unt.edu/disk2/me/ta/pt/hx/metapthx/web/4.jpg',
+                    'http://unt.edu/disk2/')
+        assert (path, location) == expected
+
+    @mock.patch('os.path.exists')
+    @mock.patch('httplib.HTTP')
+    def test_file_at_http_with_file_in_path(self, MockedHTTP, mocked_exists):
+        """Locate a file on another server via http URL when file path
+        starts with 'file://'.
+        """
+        MockedHTTP.return_value.getreply.return_value = (200, '', '')
+        mocked_exists.return_value = False
+        path, location = system.get_file_system('metapthx',
+                                                'file://web/4.jpg',
+                                                self.location_tuple)
+        expected = ('http://unt.edu/disk2/me/ta/pt/hx/metapthx/web/4.jpg',
+                    'http://unt.edu/disk2/')
+        assert (path, location) == expected
+
+    @mock.patch('os.path.exists')
+    @mock.patch('httplib.HTTP')
+    def test_file_system_with_empty_path(self, MockedHTTP, mocked_exists):
+        """Test http:// location with an '' (empty) `path`."""
+        MockedHTTP.return_value.getreply.return_value = (200, '', '')
+        mocked_exists.return_value = False
```

Comment:
In this test and the following one, you don't need to mock the `os.path.exists` function since you aren't using self.location_tuple. So I would say let's either get rid of the mock, or use the self.location_tuple. (personally prefer the latter option, but up to you)